src/test/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionListTest.java

```diff
@@ -79,4 +91,60 @@
         });
     }
 
+    @Issue("TODO")
+    @Test public void resumeStepExecutions() throws Throwable {
+        sessions.then(r -> {
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("noResume()", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            r.waitForMessage("Starting non-resumable step", b);
+            // TODO: Unclear how this might happen in practice.
+            FlowExecutionList.get().unregister(b.asFlowExecutionOwner());
```

Comment:
As mentioned, I am not sure how this might happen outside of something like a backup where files are being copied indiscriminately. The code around build completion in `WorkflowRun` looks reasonable to me from a quick glance, and the call to `unregister` is the final thing that happens, so the only thing that comes to mind is that there is some issue causing the build to not be saved as expected, for example perhaps some issue with a `BulkChange` being aborted or something like that.